portfolio/src/test/java/com/google/sps/NumCommentsTest.java

```diff
@@ -0,0 +1,200 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+ 
+import static org.mockito.Mockito.*;
+import static org.junit.Assert.assertTrue;
+
+import javax.servlet.http.*;
+import org.junit.Assert;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+
+import com.google.sps.data.Comment;
+import javax.servlet.http.*;
+import java.io.*;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@RunWith(JUnit4.class)
+public class NumCommentsTest {
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+  
+  /*
+   * Helper that simply removes all whitespace and invisible chars (tabs, \n...)
+   */
+  public String removeWhitespace(String input) {
+    return input.replaceAll("\\s+","");
+  }
+
+  /*
+   * Turn a response StringWriter into an Int
+   */
+  public int StringWriterToInt(StringWriter stringWriter) {
+    String string = stringWriter.toString();
+    String withoutWhitespace = removeWhitespace(string);
+    try {
+      int inputValue = Integer.valueOf(removeWhitespace(string));
+      return inputValue;
+    } catch (NumberFormatException nfe) {
+      throw new NumberFormatException("Couldn't parse " + withoutWhitespace);
+    }
+  } 
+
+  /*
+   * Just make a basic comment entity for testing purposes.
+   */
+  public Entity basicComment() {
```

Comment:
This is good and a very common test pattern.  You often parameterize "makeObject" methods to shorten your actual tests.